# Python Intermedio.

In [12]:
# image_rotator.py

from PIL import Image # importa el módulo Image de la librería PIL

# carga una imagen llamada 'tripleten_logo.jpeg'.
im = Image.open('tripleten_logo.png')

# obtén el tamaño de la imagen mediante el atributo .size y muéstralo
print(im.size)

# gira la imagen 90 grados en sentido contrario a las agujas del reloj
rotated = im.rotate(90)

# guarda la imagen girada
rotated.save('rotated.png')

(1110, 694)


**En este punto python ha rotado una imagen en 90 grados. EL resultado indica el 1110 (ancho) y 694 (alto).**

### **Argumentos**

Los argumentos de línea de comandos son similares a los argumentos de una función en Python. Los pasamos a nuestro script posicionalmente incluyéndolos en nuestra sentencia de ejecución de línea de comandos. Los valores de los argumentos pasados se usan dentro del script.

In [13]:
# image_rotator.py

from PIL import Image
import argparse  # importa el módulo argparse

# inicializa el analizador sintáctico
parser = argparse.ArgumentParser()

# agrega argumentos con sus nombres correspondientes
parser.add_argument("input_file")
parser.add_argument("output_file")
parser.add_argument("angle", type=int)

# analiza los argumentos
args = parser.parse_args()

# carga una imagen del argumento input_file
im = Image.open(args.input_file)

# muestra el tamaño de la imagen
print(im.size)

# gira la imagen en un ángulo proporcionado como argumento
rotated = im.rotate(args.angle)

# guarda la imagen girada usando la ruta de salida de un argumento output_file
rotated.save(args.output_file)

usage: ipykernel_launcher.py [-h] input_file output_file angle
ipykernel_launcher.py: error: the following arguments are required: input_file, output_file, angle


SystemExit: 2

In [18]:
!python image_rotator.py tripleten_logo.png output.png 180


(1110, 694)
